storm-client/src/jvm/org/apache/storm/topology/WindowedBoltExecutor.java

```diff
@@ -127,26 +154,30 @@ private void validate(Map<String, Object> topoConf, Count windowLengthCount, Dur
             throw new IllegalArgumentException("Window length is not specified");
         }
 
-        if (windowLengthDuration != null && slidingIntervalDuration != null) {
-            ensureDurationLessThanTimeout(windowLengthDuration.value + slidingIntervalDuration.value, topologyTimeout);
-        } else if (windowLengthDuration != null) {
-            ensureDurationLessThanTimeout(windowLengthDuration.value, topologyTimeout);
-        } else if (slidingIntervalDuration != null) {
-            ensureDurationLessThanTimeout(slidingIntervalDuration.value, topologyTimeout);
-        }
+        if (isPersistent()) {
+            ensureCheckpointIntervalLessThanTimeout(getCheckpointIntervalSecs(topoConf), topologyTimeout);
```

Comment:
we don't need to validate `message timeout > window length + sliding interval` here since the tuples are acked during checkpoint. Hence ensuring the `timeout > checkpoint interval` would suffice.